### PR TITLE
WIP - dev platform support

### PR DESCRIPTION
**Pull request by Berardpi**
@alexnoox  @ouranos @BrunoChauvet  if one of you could take a look
@x4d3: FYI

This is wip, still need to done some spec and testing.

I've completely changed the config vars names compared to the php and java:
Php/Java:
```php
{
  # ===> Developer Platform Configuration
  # This is the host and base path that should be used by your environment to retrieve its marketplaces configuration.
  "dev-platform": {
    "host": "https://dev-platform.maestrano.com",
    "v1_path": "/api/config/v1/marketplaces"
  },
  # => Environment credentials
  # These are your environment credentials, you can get them by connecting on the developer platform, then go on your app, they will be display under the technical view on each environment.
  "environment": {
    "name": "<your environment nid>"
    "api_key": "<your environment key>",
    "api_secret": "<your environment secret>"
  }
}
```
```
export DEVPL_HOST=<developer platform host>
export DEVPL_V1_PATH=<developer platform host>
export ENVIRONMENT_NAME=<your environment nid>
export ENVIRONMENT_KEY=<your environment key>
export ENVIRONMENT_SECRET=<your environment secret>
```

Ruby:
```
dev_platform:
  host: 'stuff'
  v1_path: 'stuff'

  environment: 'test'
  api_key: '123'
  api_secret: '123'
```
```
export MNO_DEVPL_HOST=<developer platform host>
export MNO_DEVPL_V1_PATH=<developer platform host>
export MNO_DEVPL_ENVIRONMENT_NAME=<your environment nid>
export MNO_DEVPL_KEY=<your environment key>
export MNO_DEVPL_SECRET=<your environment secret>
```

Especially for the environment variables, as the sdk is meant to be included in a larger, non maestrano project, I really think they should be prefixed. Also, maybe I missed something, but I don't get what `ENVIRONMENT_KEY`/`ENVIRONMENT_SECRET` mean (I'd understand something like `uat_key` or `prod_key` but here we can define only one)

Config file is less important, but same issue with the `environment/api_key` naming..

@alexnoox @x4d3  what do you think?

### CODE REVIEW

**Comment by x4d3** [comment]
I agree with the name changing and I think we should harmonize the SDKS and update the others.

**Comment by alexnoox** [comment]
@Berardpi I do agree on the need to prefix the variables.

There is many credentials in the dev-platform (app environment, marketplace, environment/marketplace link, tenant, user). The ones that the app provide are the "app environment" credentials, and they will find it under the environment view in the frontend. they will have a set of keys per environment and not per app.

What do you think about this? (@x4d3 @ouranos @BrunoChauvet @Berardpi)
```
export MNO_DEVPL_HOST=<developer platform host>
export MNO_DEVPL_V1_PATH=<developer platform host>
export MNO_DEVPL_ENVIRONMENT_NAME=<your environment nid>
export MNO_DEVPL_ENVIRONMENT_KEY=<your environment key>
export MNO_DEVPL_ENVIRONMENT_SECRET=<your environment secret>
```

**Comment by alexnoox** [comment]
Or this (shorter):
```
export MNO_DEVPL_HOST=<developer platform host>
export MNO_DEVPL_V1_PATH=<developer platform path>
export MNO_DEVPL_ENV_NAME=<your environment nid>
export MNO_DEVPL_ENV_KEY=<your environment key>
export MNO_DEVPL_ENV_SECRET=<your environment secret>
```

**Comment by Berardpi** [comment]
@alexnoox I'm good with that 😄

**Comment by ouranos** [comment]
I think `MNO_DEVPL_V1_PATH` is not explicit enough.

Maybe:
```
export MNO_DEVPL_CONFIG_API="/api/config/v1"
```

And add the `/marketplaces` in the code?

File: lib/maestrano/auto_configure.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      end
+
+      devpl_config = {}
+      devpl_config[:host] = ENV['MNO_DEVPL_HOST'] || yaml_config['dev_platform']['host']
+      devpl_config[:v1_path] = ENV['MNO_DEVPL_V1_PATH'] || yaml_config['dev_platform']['v1_path']
+
+      devpl_config[:env_name] = ENV['MNO_DEVPL_ENV_NAME'] || yaml_config['environment']['name']
+      devpl_config[:env_api_key] = ENV['MNO_DEVPL_ENV_KEY'] || yaml_config['environment']['api_key']
+      devpl_config[:env_api_secret] = ENV['MNO_DEVPL_ENV_SECRET'] || yaml_config['environment']['api_secret']
+
```
Comment (ouranos) [inline lib/maestrano/auto_configure.rb]: Not a biggie but why not:
```ruby
devpl_config = {
  host:    ENV['MNO_DEVPL_HOST'] || yaml_config['dev_platform']['host'],
  v1_path: ENV['MNO_DEVPL_V1_PATH'] || yaml_config['dev_platform']['v1_path'],
  ...
}
  
```

File: lib/maestrano/auto_configure.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        raise "No or bad response received from dev platform: #{e}"
+      end
+
+      response['marketplaces'].each do |marketplace|
+        Maestrano[marketplace['marketplace']].configure do |config|
+          config.environment = marketplace['environment']
+
+          [:app, :sso, :api, :webhook, :connec].each do |s|
+            (marketplace[s.to_s] || {}).each do |k, v|
+              config.send(s).send("#{k}=", v)
```
Comment (ouranos) [inline lib/maestrano/auto_configure.rb]: :neckbeard:

**Pull request merged**